<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CRCCalculator</class>
 <widget class="QWidget" name="CRCCalculator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>728</width>
    <height>349</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CRC计算器</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="6" column="1" colspan="2">
    <widget class="QLineEdit" name="lineEditInit">
     <property name="placeholderText">
      <string>例如：0xFFFF</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>初始值 INIT（Hex）</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QLineEdit" name="lineEditOutputHex"/>
   </item>
   <item row="1" column="1" colspan="5">
    <widget class="QTextEdit" name="textEdit">
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="5" column="4" colspan="2">
    <widget class="QLineEdit" name="lineEditPoly">
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>例如：0x3D65</string>
     </property>
    </widget>
   </item>
   <item row="9" column="1" colspan="5">
    <widget class="QLabel" name="labelInfo">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>获取更多crc相关信息，请访问 http://www.ip33.com/crc.html (双击访问)</string>
     </property>
    </widget>
   </item>
   <item row="5" column="3">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>多项式 POLY（Hex）</string>
     </property>
    </widget>
   </item>
   <item row="6" column="3">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>结果异或值 XOROUT（Hex）</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>参数模型 NAME</string>
     </property>
    </widget>
   </item>
   <item row="7" column="5">
    <widget class="QPushButton" name="pushButtonCopyBin">
     <property name="text">
      <string>复制</string>
     </property>
    </widget>
   </item>
   <item row="6" column="4" colspan="2">
    <widget class="QLineEdit" name="lineEditXOROUT">
     <property name="placeholderText">
      <string>例如：0xFFFF</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>校验输出（HEX）</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="5">
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QCheckBox" name="checkBoxRefIn">
        <property name="text">
         <string>输入数据反转（REFIN）</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxRefOut">
        <property name="text">
         <string>输出数据反转（REFOUT）</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonClear">
        <property name="text">
         <string>清空</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonCalculate">
        <property name="text">
         <string>计算</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1" colspan="5">
    <widget class="QLabel" name="labelInputType">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>需要校验的数据</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="5">
    <widget class="QFrame" name="frame">
     <property name="styleSheet">
      <string notr="true">QFrame {
	border:none
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QRadioButton" name="radioButtonHex">
        <property name="text">
         <string>HEX</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButtonASCII">
        <property name="text">
         <string>ASCII</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="1" colspan="4">
    <widget class="QLabel" name="label_10">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>高位在左低位在右，使用时请注意高低位顺序！！！</string>
     </property>
    </widget>
   </item>
   <item row="7" column="4">
    <widget class="QLineEdit" name="lineEditOutputBin"/>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>宽度 WIDTH</string>
     </property>
    </widget>
   </item>
   <item row="7" column="2">
    <widget class="QPushButton" name="pushButtonCopyHex">
     <property name="text">
      <string>复制</string>
     </property>
    </widget>
   </item>
   <item row="7" column="3">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>校验输出（BIN）</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1" colspan="2">
    <widget class="QComboBox" name="comboBoxWidth"/>
   </item>
   <item row="4" column="1" colspan="2">
    <widget class="QComboBox" name="comboBoxName"/>
   </item>
   <item row="4" column="3">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>多项式公式</string>
     </property>
    </widget>
   </item>
   <item row="4" column="4" colspan="2">
    <widget class="QLabel" name="labelPolyFormula">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>计算面板</string>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>label_6</zorder>
  <zorder>lineEditOutputHex</zorder>
  <zorder>textEdit</zorder>
  <zorder>lineEditPoly</zorder>
  <zorder>labelInfo</zorder>
  <zorder>label_5</zorder>
  <zorder>label_7</zorder>
  <zorder>label</zorder>
  <zorder>pushButtonCopyBin</zorder>
  <zorder>lineEditXOROUT</zorder>
  <zorder>label_8</zorder>
  <zorder>frame_2</zorder>
  <zorder>comboBoxName</zorder>
  <zorder>labelInputType</zorder>
  <zorder>label_3</zorder>
  <zorder>frame</zorder>
  <zorder>label_10</zorder>
  <zorder>lineEditOutputBin</zorder>
  <zorder>label_2</zorder>
  <zorder>pushButtonCopyHex</zorder>
  <zorder>label_9</zorder>
  <zorder>comboBoxWidth</zorder>
  <zorder>lineEditInit</zorder>
  <zorder>label_4</zorder>
  <zorder>labelPolyFormula</zorder>
  <zorder>label_11</zorder>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonClear</sender>
   <signal>clicked()</signal>
   <receiver>textEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>578</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>560</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCopyHex</sender>
   <signal>clicked()</signal>
   <receiver>lineEditOutputHex</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCopyHex</sender>
   <signal>clicked()</signal>
   <receiver>lineEditOutputHex</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCopyBin</sender>
   <signal>clicked()</signal>
   <receiver>lineEditOutputBin</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>622</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCopyBin</sender>
   <signal>clicked()</signal>
   <receiver>lineEditOutputBin</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonHex</sender>
   <signal>clicked()</signal>
   <receiver>textEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
